internal/services/redis/redis_cache_access_policy_assignment_resource.go

```diff
@@ -0,0 +1,213 @@
+package redis
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/redis/2023-08-01/redis"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type RedisCacheAccessPolicyAssignmentResource struct {
+}
+
+var _ sdk.ResourceWithUpdate = RedisCacheAccessPolicyAssignmentResource{}
+
+type RedisCacheAccessPolicyAssignmentResourceModel struct {
+	Name             string `tfschema:"name"`
+	RedisCacheID     string `tfschema:"redis_cache_id"`
+	AccessPolicyName string `tfschema:"access_policy_name"`
+	ObjectID         string `tfschema:"object_id"`
+	ObjectIDAlias    string `tfschema:"object_id_alias"`
+}
+
+func (r RedisCacheAccessPolicyAssignmentResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"redis_cache_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+		},
+		"access_policy_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"object_id": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
+		"object_id_alias": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+		},
```

Comment:
could we add validation here for the possible values `ServicePrincipal` and `UserMSI`?

Reply:
@catriona-m I am curious of what the purpose for this validation, I first created the resources with azapi and am now trying to import them to the azurerm provider

`object_id_alias    = "xxx@yyy.com" -> "UserMSI" # forces replacement`

Ref. https://learn.microsoft.com/en-us/azure/templates/microsoft.cache/redis/accesspolicyassignments?pivots=deployment-language-terraform#rediscacheaccesspolicyassignmentproperties-2

Any insights you could share?

Reply:
Thanks for highlighting this @audunsolemdal, it looks like we'll need to update the validation to account for usernames too.

Reply:
In my case I also set object_id_alias =  <name of my app service>

We are yet to test redis cache policies from these app services though. Not sure how exactly this attribute is intended to be used

Reply:
@audunsolemdal @catriona-m ahh, so it *is* an arbitrary string. I'll add a PR to make it StringNotEmpty validation